### ISSUE TITLE
RadiativeTransferOpticallThinSourceClustering does not work

### ISSUE THREAD

**Issue by aemerick** [enzo-project/enzo-dev#147]
Although this did seem to be working in my tests in PR#127 (in an isolated galaxy set-up), I've found it fails miserably in a cosmological simulation. Its not clear to me what exactly is going wrong, but it is clear that this does not work well in all cases. 

In some instances it appeared that the box was filled with a near-uniform radiation flux from the source clustering. In others there would be large-scale (most of the box) circular caustic-like features. I varied `RadiativeTransferPhotonMergeRadius` from 2 to 15 to try and see if that affected anything, but did not seem to find any value that worked well.

I'll try and update this if I figure out the issue, but last time I tried I had a hard time digging into the code.

**Comment by aemerick** [comment]
I did some more digging into this. The issue isn't as dramatic as I first thought. Part of my issue was a units problem, but now that I've tracked that down, the issue mostly goes away depending on choice of `RadiativeTransferPhotonMergeRadius`. In my tests, it looks like a small value for this (~2) does much better than a larger value (~10). And I know this has come up before, but I don't quite have a good intuition for what values for this actually mean in practice (@jwise77 ?)....

But in my low-res tests (a 1 Mpc co-moving box with 29 sources at 5 levels of refinement, with each source on level 5), it looks like this does very well generally. To pick a random example:

![FUV_profile_16777237](https://user-images.githubusercontent.com/6494319/82250076-b17c0980-98ff-11ea-9e70-d220663b8f1d.png)

The clustering lines and the direct method lines lie all on top of each other (black dashed is just there for reference) for small values of the merge radius. Having `RadiativeTransferPhotonMergeRadius` set to 10 misses the radiation in every case. 

However, there are a couple instances where things go wrong (example 1):

![FUV_profile_16777249](https://user-images.githubusercontent.com/6494319/82250165-d1133200-98ff-11ea-819c-ba7a68119d73.png)

Where the tests with smaller values of the merge radius are a constant factor (about 0.2) of the direct method. Of the 29 particles, this happens for 3 of the particles, all three of which are < 1 grid cell apart from each other (however, other particles with similar separations do OK).  

And in one case, all methods miss the radiation entirely (example 2):

![FUV_profile_16777226](https://user-images.githubusercontent.com/6494319/82250400-3bc46d80-9900-11ea-9889-061c0c6aeb63.png)

This particle does have something unique, in that it is very isolated, and the only radiation source for 10 kpc (the rest have nearest-neighbor distances of a few to 10 pc). 

In summary, of the 29 particles in this test, 4 have issues in the optically thin source clustering method. However, I don't know how typical that is (I'm not sure if more sources have issues if I check a few cycles later) since I'm not really sure what situations give rise to the problem.

**Comment by jwise77** [comment]
Having higher R values giving worse results goes against my initial expectations because I'm so used associating the merge radius parameter to ray merging. For the rays, you get better results with larger merging radii.

But with this tree code, it should have a similar behavior. There's a definitely a bug here. I can imagine with the isolated particle in the last case, the lack of flux is coming from an incorrect pairing or tree construction. Near the isolated particle (r < R*separation), the flux calculation should be a direct 1/r^2 from the particle itself. It seems like the flux is only coming from the distant particles. Maybe you can use this last case to see why the particle isn't included in the calculation, and solving this might help the other cases.
